enumerable.rb

```diff
@@ -0,0 +1,116 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    var = to_a
+    result = []
+    i = 0
+    while i < var.length
+      result << yield(var[i])
+      i += 1
+    end
+    result
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    var = to_a
+    result = i = 0
+    while i < var.length
+      result = yield(var[i], i)
+      i += 1
+    end
+    result
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    my_each { |i| arr.push(i) if yield(i) }
+    arr
+  end
+
+  def my_all?
+    var = to_a
+    var.empty?
+    if block_given?
+      my_each { |i| return false unless yield(i) }
+      return true
+    end
+    var.include?(nil) || var.include?(false) ? false : true
+  end
+
+  def my_any?
+    var = to_a
+    if block_given?
+      var.my_each { |i| return true if yield(i) }
+    else
+      var.my_each { |i| return true if i.nil? == false and i != false }
+    end
+    false
+  end
+
+  def my_none?
+    var = to_a
+    if block_given?
+      var.my_each { |i| return false if yield(i) }
+    else
+      var.my_each { |i| return false if i.nil? == false and i != false }
+    end
+    true
+  end
+
+  def my_count(*arg)
+    sum = 0
+    if block_given?
+      my_each { |i| sum += 1 if yield(i) }
+      sum
+    elsif arg.empty?
+      length
+    else
+      my_each { |i| sum += 1 if arg[0] == i }
+      sum
+    end
+  end
+
+  def my_map(arg = nil)
+    arr = []
+    if arg
+      my_each { |i| arr.push(arg.call(i)) }
+      return arr
+    end
+    return unless block_given?
+
+    my_each { |i| arr.push(yield(i)) }
+    arr
+  end
+
+  def my_inject(*arg)
+    memo = 0
+    if arg.empty?
+      my_each { |i| memo = yield(memo, i) }
+      return memo
+    end
+    if arg.empty? == false && block_given?
+      my_each do |i|
+        memo = yield(arg[0], i)
+        arg[0] = memo
+      end
+      return memo
+    end
+    if arg[0].is_a? Symbol
+      my_each { |i| memo = memo.send(arg[0], i) }
+      return memo
+    end
+    return unless arg.length == 2
+
+    my_each { |i| memo = arg[0] = arg[0].send(arg[1], i) }
+    memo
+  end
```

Comment:
- [ ] should have working `my_inject`  identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

####  my_inject
> Remember to test it not only for arrays but for ranges as well

- [ ] when a block is given without an initail value combines all elements of enum by applying a binary operation, specified by a block::range
```ruby
(1..5).my_inject { |prod, n| prod * n } == (1..5).inject { |prod, n| prod * n } => 120
```
- Yours
![image](https://user-images.githubusercontent.com/52209067/94222668-3cbb6c00-feab-11ea-8746-fc228f568a12.png)
 retrieves the wrong answer

- Real inject
![image](https://user-images.githubusercontent.com/52209067/94222679-4644d400-feab-11ea-80b8-020fe618e027.png)

- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.
```ruby
(1..5).my_inject(:*) == (1..5).inject(:*) => 120
```
- Yours retrieves the wrong answer
![image](https://user-images.githubusercontent.com/52209067/94222778-90c65080-feab-11ea-9864-a1393ff0b9df.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/94222810-9f146c80-feab-11ea-8f93-fa97ed4d387e.png)